File: apps/hook_frontend/src/types/affiliate.ts
export type AdminAffiliateUserSummary = {
  id: string;
  username: string;
  email: string;
  affiliate_code: string;
};

export type AdminAffiliateOverviewResponse = {
  total_referred_users: number;
  active_referrer_count: number;
  total_commission_amount: number;
  today_commission_amount: number;
  month_commission_amount: number;
  affiliate_commission_percent: number;
};

export type AdminAffiliateRelation = {
  user: AdminAffiliateUserSummary;
  referrer: AdminAffiliateUserSummary | null;
  referred_at: string | null;
  referred_recharge_amount: number;
  commission_amount: number;
  last_commission_at: string | null;
};

export type AdminAffiliateRelationListResponse = {
  items: AdminAffiliateRelation[];
  total: number;
  page: number;
  page_size: number;
};

export type AdminAffiliateRelationChange = {
  id: string;
  user: AdminAffiliateUserSummary;
  old_referrer: AdminAffiliateUserSummary | null;
  new_referrer: AdminAffiliateUserSummary | null;
  operator: AdminAffiliateUserSummary | null;
  operator_user_id: string | null;
  reason: string;
  created_at: string;
};

export type AdminAffiliateRelationChangeListResponse = {
  items: AdminAffiliateRelationChange[];
  total: number;
  page: number;
  page_size: number;
};

export type AdminAffiliateRelationUpdateInput = {
  referrer_aff_code?: string;
  clear_referrer?: boolean;
  reason: string;
};

export type AdminAffiliateCommission = {
  id: string;
  referrer: AdminAffiliateUserSummary;
  referred: AdminAffiliateUserSummary;
  recharge_order_id: string;
  recharge_order_no: string;
  payable_amount: number;
  commission_percent: number;
  commission_amount: number;
  wallet_transaction_id: string | null;
  status: string;
  failure_reason: string | null;
  created_at: string;
};

export type AdminAffiliateCommissionListResponse = {
  items: AdminAffiliateCommission[];
  total: number;
  page: number;
  page_size: number;
};

export type AdminAffiliateDailyReportItem = {
  date: string;
  commission_order_count: number;
  referred_payer_count: number;
  payable_amount: number;
  commission_amount: number;
};

export type AdminAffiliateReferrerReportItem = {
  referrer: AdminAffiliateUserSummary;
  referred_user_count: number;
  commission_order_count: number;
  payable_amount: number;
  commission_amount: number;
};

export type AdminAffiliateReportResponse = {
  daily_items: AdminAffiliateDailyReportItem[];
  referrer_items: AdminAffiliateReferrerReportItem[];
  referrer_total: number;
  page: number;
  page_size: number;
};

File: crates/types/src/user/affiliate.rs
use rust_decimal::Decimal;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Serialize)]
pub struct AffiliateSummaryResponse {
    pub affiliate_code: String,
    pub affiliate_link: String,
    pub affiliate_enabled: bool,
    pub referred_user_count: u64,
    #[serde(with = "rust_decimal::serde::float")]
    pub total_referred_recharge_amount: Decimal,
    #[serde(with = "rust_decimal::serde::float")]
    pub total_commission_amount: Decimal,
    #[serde(with = "rust_decimal::serde::float")]
    pub today_commission_amount: Decimal,
    #[serde(with = "rust_decimal::serde::float")]
    pub month_commission_amount: Decimal,
    #[serde(with = "rust_decimal::serde::float")]
    pub affiliate_commission_percent: Decimal,
    pub last_commission_at: Option<String>,
}

#[derive(Clone, Debug, Deserialize, PartialEq, Eq)]
pub struct AffiliateReferralQuery {
    pub page: u64,
    pub page_size: u64,
    #[serde(default)]
    pub search: Option<String>,
    #[serde(default)]
    pub referred_start: Option<String>,
    #[serde(default)]
    pub referred_end: Option<String>,
}

#[derive(Clone, Debug, PartialEq, Serialize)]
pub struct AffiliateReferralItem {
    pub referred_user_id: String,
    pub username: String,
    pub masked_email: String,
    pub referred_at: Option<String>,
    #[serde(with = "rust_decimal::serde::float")]
    pub referred_recharge_amount: Decimal,
    #[serde(with = "rust_decimal::serde::float")]
    pub commission_amount: Decimal,
    pub last_commission_at: Option<String>,
}

#[derive(Clone, Debug, PartialEq, Serialize)]
pub struct AffiliateReferralListResponse {
    pub items: Vec<AffiliateReferralItem>,
    pub total: u64,
    pub page: u64,
    pub page_size: u64,
}

#[derive(Clone, Debug, Deserialize, PartialEq)]
pub struct AffiliateCommissionQuery {
    pub page: u64,
    pub page_size: u64,
    #[serde(default)]
    pub referred_search: Option<String>,
    #[serde(default)]
    pub recharge_order_no: Option<String>,
    #[serde(default)]
    pub start_at: Option<String>,
    #[serde(default)]
    pub end_at: Option<String>,
    #[serde(default, with = "rust_decimal::serde::float_option")]
    pub min_commission_amount: Option<Decimal>,
    #[serde(default, with = "rust_decimal::serde::float_option")]
    pub max_commission_amount: Option<Decimal>,
}

#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
pub struct AffiliateReferredUserSummary {
    pub referred_user_id: String,
    pub username: String,
    pub masked_email: String,
}

#[derive(Clone, Debug, PartialEq, Serialize)]
pub struct AffiliateCommissionItem {
    pub id: String,
    pub referred: AffiliateReferredUserSummary,
    pub recharge_order_no: String,
    #[serde(with = "rust_decimal::serde::float")]
    pub payable_amount: Decimal,
    #[serde(with = "rust_decimal::serde::float")]
    pub commission_percent: Decimal,
    #[serde(with = "rust_decimal::serde::float")]
    pub commission_amount: Decimal,
    pub wallet_transaction_id: Option<String>,
    pub status: String,
    pub failure_reason: Option<String>,
    pub created_at: String,
}

#[derive(Clone, Debug, PartialEq, Serialize)]
pub struct AffiliateCommissionListResponse {
    pub items: Vec<AffiliateCommissionItem>,
    pub total: u64,
    pub page: u64,
    pub page_size: u64,
}
